### PR TITLE
[AUD-28] Bust web3 provider cache on balance of

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Description
Invalidates the cache after local state changes to claim amt and balance.

Currently the web3 provider can cache requests to your current balance and if we poll for your balance and get a cached value right after a local state change, the value can toggle back to what it was before your local state change.

The AudiusTokenClient .bustCache method invalidates the cache by adding a nonce query param -- this feature could be extended to the ContractClient.js class in libs at some point if we need this behavior more than for fetching balance

### Dragons

Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?
Nah

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.

Ran against prod & issued a claim. Ensured that numbers were consistent.

### CODE REVIEW

**Comment by raymondjacobson** [comment]
LMK if you can think of a cleaner way to do this

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/rj-balanceof-bust-cache

File: src/store/wallet/slice.ts [hunk 1]
```diff
@@ -20,11 +20,13 @@ export type WalletAddress = string
...
   balance: Nullable<StringWei>
   pendingClaimBalance: Nullable<StringWei>
+  localBalanceChange: boolean
 }
 
 const initialState: WalletState = {
   balance: null,
-  pendingClaimBalance: null
+  pendingClaimBalance: null,
+  localBalanceChange: false
```
Comment (piazzatron) [inline src/store/wallet/slice.ts:29]: nit: wdyt about calling it something like `localBalanceDidChange`? When I first saw at it I assumed this was going to be a numeric value

**Review by piazzatron — APPROVED** [review]
LGTM

File: src/store/wallet/sagas.ts [hunk 2]
```diff
@@ -113,13 +114,10 @@ function* getWalletBalanceAndClaim() {
 }
 
 function* fetchBalanceAsync() {
-  const oldBalance: ReturnType<typeof getAccountBalance> = yield select(
-    getAccountBalance
-  )
+  const localBalanceChange = yield select(getLocalBalanceChange)
```
[the remark this answers is not in the document]
Comment (raymondjacobson) [reply src/store/wallet/sagas.ts:117]: no reason, fixed :)

**Pull request merged**